File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
        StringBuilder ans = new StringBuilder();

		for (int i = 0; i < arr.length; i++) {
			if (flag[i]) ans.append(String.valueOf(arr[i] + "&").repeat(arr[i] * 2));
			else ans.delete(ans.length() - arr[i] * 2, ans.length());
		}

		return Arrays.stream(ans.toString().split("&")).mapToInt(Integer::parseInt).toArray();
    }
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.kt
import java.util.Stack

class Solution {
    fun solution(arr: IntArray, flag: BooleanArray): IntArray {
        val ans = Stack<Int>()
        for (i in arr.indices) {
            for (j in 0 .. (arr[i] * if (flag[i]) 2 else 1) - 1) {
                if (flag[i]) ans.push(arr[i]) else ans.pop()
            }
        }
        
        return ans.toIntArray()
    }
}